features/storage/kvstore/tdbstore/DirectAccessDevicekey.h

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_DIRECT_ACCESS_DEVICEKEY_H
+#define MBED_DIRECT_ACCESS_DEVICEKEY_H
+
+#include "FlashIAP.h"
+#include <stdio.h>
+
+int direct_access_to_devicekey(uint32_t tdb_start_offset, uint32_t tdb_end_offset, void *data_buf,
```

Comment:
is this missing doxygen ?

Reply:
sorry, I added doxygen info - and squashed the log

Reply:
Thanks, looks better. But still can be improved: "inital version" - what initial version is it? what does it bring? or `successfully compiled` ?

Reply:
@0xc0170 - I'm not sure I follow.. git is source control. Part of its use is to track the different development (e.g. enabling quick recovery milestones..) . I agree that sometimes there are too many unimportant stages that can be squashed .. but in general I don't think we should rewrite git-log history prior to delivery or make it look better)..

Reply:
I partly agree but my point was that this feels like development branch rather than ready for the release. Don't need to squash but rather add details important for users.
Two commits msg have zero value without studying changes and this is functionality change, we should introduce it - why are we bringing this functionality change.

Reply:
Thanks, I think I now understand what you meant - I'll add the required information.